maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -47,26 +56,69 @@ public OpenTelemetrySdkService() {
         "OpenTelemetry: Initialize OpenTelemetrySdkService v{}...",
         MavenOtelSemanticAttributes.TELEMETRY_DISTRO_VERSION_VALUE);
 
-    // Change default of "otel.[traces,metrics,logs].exporter" from "otlp" to "none"
-    // The impacts are
-    // * If no otel exporter settings are passed, then the Maven extension will not export
-    //   rather than exporting on OTLP GRPC to http://localhost:4317
-    // * If OTEL_EXPORTER_OTLP_ENDPOINT is defined but OTEL_[TRACES,METRICS,LOGS]_EXPORTER,
-    //   is not, then don't export
-    Map<String, String> properties = new HashMap<>();
-    properties.put("otel.traces.exporter", "none");
-    properties.put("otel.metrics.exporter", "none");
-    properties.put("otel.logs.exporter", "none");
-
     AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk =
         AutoConfiguredOpenTelemetrySdk.builder()
             .setServiceClassLoader(getClass().getClassLoader())
-            .addPropertiesSupplier(() -> properties)
+            .addPropertiesCustomizer(
+                configProperties -> {
+                  // The OTel SDK by default sends data to the OTLP gRPC endpoint at localhost:4317.
+                  // Change this behavior to disable by default the OTel SDK in the Maven extension
+                  // so
+                  // that it must be explicitly enabled by the user.
+                  // To change this default behavior, we set "otel.[traces,metrics,logs].exporter"
+                  // to
+                  // "none" if the endpoint has not been specified
+                  if (configProperties.getString("otel.exporter.otlp.endpoint") == null) {
+                    Map<String, String> properties = new HashMap<>();
+                    if (Objects.equals(
+                            "otlp", configProperties.getString("otel.traces.exporter", "otlp"))
+                        && configProperties.getString("otel.exporter.otlp.traces.endpoint")
+                            == null) {
+                      properties.put("otel.traces.exporter", "none");
+                    }
+                    if (Objects.equals(
+                            "otlp", configProperties.getString("otel.metrics.exporter", "otlp"))
+                        && configProperties.getString("otel.exporter.otlp.metrics.endpoint")
+                            == null) {
+                      properties.put("otel.metrics.exporter", "none");
+                    }
+                    if (Objects.equals(
+                            "otlp", configProperties.getString("otel.logs.exporter", "otlp"))
+                        && configProperties.getString("otel.exporter.otlp.logs.endpoint") == null) {
+                      properties.put("otel.logs.exporter", "none");
+                    }
```

Comment:
[minor] it would be more readable to create a helper method to check this as it takes quite an effort to read and understand it that we check two config options to not be explicitly set and set one of them in the modified config, all of that 3 times with close but slightly different variables for logs, traces and metrics.

Reply:
Please review the helper method I introduced